sapi/cli/php_cli_server.c

```diff
@@ -365,7 +365,14 @@ static void append_essential_headers(smart_str* buffer, php_cli_server_client *c
 
 static const char *get_mime_type(const php_cli_server *server, const char *ext, size_t ext_len) /* {{{ */
 {
-	return (const char*)zend_hash_str_find_ptr(&server->extension_mime_types, ext, ext_len);
+	char *ret;
+	ALLOCA_FLAG(use_heap)
+	char *ext_lower = do_alloca(ext_len + 1, use_heap);
+	memcpy(ext_lower, ext, ext_len + 1);
+	zend_str_tolower(ext_lower, ext_len);
```

Comment:
You can use zend_str_tolower_copy and drop the separate memcpy.

Reply:
Ah, nice. Thanks!